text/0000-const-ub.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: `const_ub`
+- Start Date: 2020-10-10
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Define how UB during const evaluation is treated:
+some kinds of UB must be detected, the remaining UB conditions are ignored and evaluation continues in a well-defined way.
+However, CTFE queries causing UB are not subject to stability guarantees and thus may fail to build in the future (e.g. when more UB is being detected).
+
+# Motivation
+[motivation]: #motivation
+
+So far, nothing is specified about what happens when `unsafe` code leads to UB during CTFE.
+This is a major blocker for stabilizing `unsafe` operations in const-contexts.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+There are some values that Rust needs to compute at compile-time.
+This includes the initial value of a `const`/`static`, and array lengths (and more general, const generics).
+Computing these initial values is called compile-time function evaluation (CTFE).
+CTFE in Rust is very powerful and permits running almost arbitrary Rust code.
+This begs the question, what happens when there is `unsafe` code and it causes [Undefined Behavior (UB)][UB]?
```

Comment:
nit: "This raises the question.", not [begs the question](https://en.wikipedia.org/wiki/Begging_the_question).

Reply:
Yeah I felt I have seen "begs the question" before in various contexts, but I can still easily fix this to "raises the question" and try to remember this for my future writing. ;)

Reply:
> I forget where I read it or I'd link it, but there's a strong argument that, in cases like these, the battle has already been lost and it does nobody any good to fight English's natural evolution toward simplicity and intuitiveness.

I claim that in clear writing, one should simply avoid the phrase entirely. Otherwise you're going to confuse people who *do* associate the phrase with its original meaning.

Reply:
In a formal document, meant to last, of course. You want to avoid ambiguity wherever possible... but, in less formal or more transient contexts, people are *going* to use it that way.

Had you said something like "nit: 'begs the question' is ambiguous. Should be 'raises the question'." so your intent in saying it wasn't ambiguous, I wouldn't have had a problem with it.